### PR TITLE
Controlled shuffles mapitop2

### PR DESCRIPTION
**Pull request by jonathanAK**
added mapitop encoding

### CODE REVIEW

File: IgOmeProfiling_pipeline.py [hunk 1]
```diff
@@ -190,7 +191,7 @@ def run_pipeline(fastq_path, barcode2samplename_path, samplename2biologicalcondi
                  args.left_construct, args.right_construct, args.max_mismatches_allowed, args.min_sequencing_quality, True if args.gz else False,
                  args.max_msas_per_sample, args.max_msas_per_bc,
                  args.max_number_of_cluster_members_per_sample, args.max_number_of_cluster_members_per_bc,
-                 args.allowed_gap_frequency, concurrent_cutoffs, args.meme_split_size, args.number_of_random_pssms,
+                 args.allowed_gap_frequency, concurrent_cutoffs, args.meme_split_size, True if args.mapitope else False, args.number_of_random_pssms,
```
Comment (shacharmo) [inline IgOmeProfiling_pipeline.py]: The argument type should be boolean, no need for condition

File: reads_filtration/mapitope_conversion.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    args = parser.parse_args()
+
+    import logging
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger('main')
+
+    mapitope_convert_all_samples(args.reads_filtration_path, args.barcode2samplename_path, args.done_file_path,sys.argv)
```
Comment (shacharmo) [inline reads_filtration/mapitope_conversion.py]: Add new line at end of file, it is considered best practice in Python

File: reads_filtration/module_wraper.py [hunk 3]
```diff
@@ -57,7 +83,7 @@ def run_first_phase(fastq_path, first_phase_output_path, logs_dir, barcode2sampl
                 continue
             for file in os.listdir(dir_path):
                 # look for faa files to collapse
-                if not file.startswith(f'{dir_name}.faa'):  # maybe there's a .gz afterwards
+                if not file.startswith(f'{dir_name}.faa') and not file.startswith(f'{dir_name}_mapitope.faa'):  # maybe there's a .gz afterwards
```
Comment (shacharmo) [inline reads_filtration/module_wraper.py:89]: We also want unique rpm for mapitope

File: reads_filtration/mapitope_conversion.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+def convert_faa_file_by_maptope(input_file_path, output_file_path):
+    with open(input_file_path,'r') as input_file, open(output_file_path,'w+') as output_file:
+        input_lines = input_file.readlines()
+        for line in input_lines:
+            if line[0] == '>':
+                output_file.write(line)  
+            else:
+                output_file.write(convert_to_mapitope(line[:-1])+'\n')
+
+def mapitope_convert_all_samples(reads_filtration_path, barcode2samplename_path, done_file_path, argv='no_argv'):
```
Comment (shacharmo) [inline reads_filtration/mapitope_conversion.py]: Can be done in parallel as at this point we have a separate file/directory per sample

**Review by shacharmo — CHANGES_REQUESTED** [review]
Overall very good.

Phase 3 (model fitting) also uses faa files and should support mapitope

File: reads_filtration/module_wraper.py (hunk 3 above)
Comment (jonathanAK) [reply reads_filtration/module_wraper.py:89]: this will create unique rpms for mapitop files.
or do you mean rpm_factors.txt?

**Review by shacharmo — APPROVED** [review]

**Pull request merged**